### PR TITLE
Check two available nodes first for running sctp

### PR DESCRIPTION
**Pull request by weliang1**
1. This PR is try to fix the issue in https://issues.redhat.com/browse/OCPQE-2834
2. Old feature/test/sctp.feature include the debug steps and debug logs from CI testing showed the testing failed is because sometimes there are less than two ready and schedulable workers in the CI cluster. The debug purpose of feature/test/sctp.feature is complete, remove feature/test/sctp.feature from this PR
3. Update feature/network/sctp.feature to have a step to check whether the cluster has at least two ready and schedulable workers.
4. Add more options for nc to make script more reliable.

Test logs: 
https://mastern-jenkins-csb-openshift-qe.cloud.paas.psi.redhat.com/job/Runner-v3-smoke/2845/console

@zhaozhanqi @huiran0826 @anuragthehatter @rbbratta @asood @brahaney @kedark3

### CODE REVIEW

File: features/networking/sctp.feature [hunk 1]
```diff
@@ -26,48 +27,45 @@ Feature: SCTP related scenarios
...
-      | oc_opts_end      |                                                                  |
-      | exec_command     | bash                                                             |
-      | exec_command_arg | -c                                                               |
-      | exec_command_arg | echo test-openshift \| nc -v <%= cb.serverpod_ip %> 30102 --sctp |
+      | pod              | sctpclient                                                            |
+      | namespace        | <%= project.name %>                                                   |
+      | oc_opts_end      |                                                                       |
+      | exec_command     | bash                                                                  |
+      | exec_command_arg | -c                                                                    |
+      | exec_command_arg | echo test-openshift \| nc -w 5 -v <%= cb.serverpod_ip %> 30102 --sctp |
```
Comment (liangxia) [inline features/networking/sctp.feature:49]: We can use 
```
When I execute on the "xxx" pod:
  | blabla |
```

File: features/networking/sctp.feature [hunk 2]
```diff
@@ -26,48 +27,45 @@ Feature: SCTP related scenarios
       | ["spec"]["nodeName"]      | <%= cb.workers[1].name %> |
     Then the step should succeed
     And the pod named "sctpclient" becomes ready
-
+
```
Comment (liangxia) [inline features/networking/sctp.feature:30]: This might be an unexpected change.

File: features/networking/sctp.feature [hunk 3]
```diff
@@ -90,47 +88,44 @@ Feature: SCTP related scenarios
...
-    """"
+    """
 
   # @author weliang@redhat.com
   # @case_id OCP-28759
   @admin
   @destructive
-  Scenario: Expose SCTP NodePort Services
-    Given I store the workers in the :workers clipboard
+  Scenario: Expose SCTP NodePort Services
```
Comment (liangxia) [inline features/networking/sctp.feature:119]: White spaces added at end of line.

Comment (weliang1) [reply features/networking/sctp.feature]: Fixed it, thanks!

File: features/networking/sctp.feature (hunk 2 above)
Comment (weliang1) [reply features/networking/sctp.feature:30]: @liangxia Sorry, not get your point here, could you explain more specifically?

File: features/networking/sctp.feature (hunk 1 above)
Comment (weliang1) [reply features/networking/sctp.feature]: @liangxia Your suggestion makes steps more short now, thanks! New test log: https://mastern-jenkins-csb-openshift-qe.cloud.paas.psi.redhat.com/job/Runner-v3-smoke/2849/consoleFull

File: features/networking/sctp.feature (hunk 2 above)
Comment (liangxia) [reply features/networking/sctp.feature:30]: White spaces introduced on the line.

File: features/networking/sctp.feature [hunk 4]
```diff
@@ -153,46 +128,33 @@ Feature: SCTP related scenarios
...
-      | pod              | sctpserver          |
-      | namespace        | <%= project.name %> |
-      | oc_opts_end      |                     |
-      | exec_command     | bash                |
-      | exec_command_arg | -c                  |
-      | exec_command_arg | nc -l 30102 --sctp  |
-    Then the step should succeed
-    """
+   When I execute on the "sctpserver" pod:
+      | bash | -c  | nc -k -l 30102 --sctp |
```
Comment (liangxia) [inline features/networking/sctp.feature:133]: Indention issue for those two lines.

File: features/networking/sctp.feature [hunk 5]
```diff
@@ -209,33 +171,19 @@ Feature: SCTP related scenarios
...
-      | pod              | sctpserver          |
-      | namespace        | <%= project.name %> |
-      | oc_opts_end      |                     |
-      | exec_command     | bash                |
-      | exec_command_arg | -c                  |
-      | exec_command_arg | nc -l 30102 --sctp  |
-    Then the step should succeed
-    """
+     When I execute on the "sctpserver" pod:
+      | bash | -c  | nc -k -l 30102 --sctp |
```
Comment (liangxia) [inline features/networking/sctp.feature:175]: Indention issue.

File: features/networking/sctp.feature [hunk 6]
```diff
@@ -262,30 +208,16 @@ Feature: SCTP related scenarios
...
-      | namespace        | <%= project.name %> |
-      | oc_opts_end      |                     |
-      | exec_command     | bash                |
-      | exec_command_arg | -c                  |
-      | exec_command_arg | nc -l 30102 --sctp  |
-    Then the step should succeed
-    """
-
+     When I execute on the "sctpserver" pod:
+      | bash | -c  | nc -k -l 30102 --sctp |
```
Comment (liangxia) [inline features/networking/sctp.feature:212]: Indention issue

**Comment by akostadinov** [comment]
Do you remove scenarios intentionally? If so LGTM from my end when existing comments are addressed.

File: features/networking/sctp.feature (hunk 6 above)
Comment (weliang1) [reply features/networking/sctp.feature:212]: Fixed, thanks!

File: features/networking/sctp.feature (hunk 2 above)
Comment (weliang1) [reply features/networking/sctp.feature:30]: Fixed, thanks!

File: features/networking/sctp.feature (hunk 4 above)
Comment (weliang1) [reply features/networking/sctp.feature:133]: Fixed, thanks!

File: features/networking/sctp.feature (hunk 5 above)
Comment (weliang1) [reply features/networking/sctp.feature:175]: Fixed, thanks!

**Comment by weliang1** [comment]
> Do you remove scenarios intentionally? If so LGTM from my end when existing comments are addressed.

@akostadinov Yes, I did intentionally

**Comment by zhaozhanqi** [comment]
/lgtm cancel
@weliang1 one question , I saw your store the ready or schedule worker,  However, `And I install machineconfigs load-sctp-module` this step if still affect all of worker even it's schduledisable?

File: features/networking/sctp.feature [hunk 7]
```diff
@@ -5,13 +5,14 @@ Feature: SCTP related scenarios
   @admin
   @destructive
   Scenario: Establish pod to pod SCTP connections
-    Given I store the workers in the :workers clipboard
-    Given I install machineconfigs load-sctp-module
-    Given I have a project
+    Given I store the ready and schedulable workers in the :workers clipboard
```
Comment (zhaozhanqi) [inline features/networking/sctp.feature:8]: hmm, this step can make sure we have 2 or more schedulable workers? I think it's only store the ready and schedulable worker.

Comment (weliang1) [reply features/networking/sctp.feature:8]: @zhaozhanqi https://github.com/openshift/verification-tests/pull/1823/files#diff-7539dbdee665a3ee90dc338a1a8258f9c4723c11850769dfbf6216c6e55a5f42R1178 will make sure the cluster will have more than one ready and schedulable worker.

**Comment by weliang1** [comment]
> /lgtm cancel
> @weliang1 one question , I saw your store the ready or schedule worker, However, `And I install machineconfigs load-sctp-module` this step if still affect all of worker even it's schduledisable?

@zhaozhanqi the step of "load-sctp-module" will not load sctp module in any  schduledisable workers

File: features/networking/sctp.feature (hunk 7 above)
Comment (zhaozhanqi) [reply features/networking/sctp.feature:8]: ah, ok

**Pull request merged**